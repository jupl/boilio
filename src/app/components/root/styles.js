/** Styling for root page */
export const PAGE = {
  width: '100%',
  height: '100%',
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  background: 'linear-gradient(white, gainsboro)',
}
