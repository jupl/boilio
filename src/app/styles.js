/** Styling for app container */
export const CONTAINER_STYLE = {
  position: 'fixed',
  top: 0,
  bottom: 0,
  left: 0,
  right: 0,
  overflow: 'auto',
  display: 'flex',
  backgroundColor: 'white',
}
