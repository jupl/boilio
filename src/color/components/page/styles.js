/** Styling for container background */
export const BACKGROUND = {
  flexDirection: 'row',
  justifyContent: 'center',
  alignItems: 'center',
}

/** Styling for text */
export const TEXT = {
  fontSize: 24,
}

/** Styling for buttons */
export const BUTTON = {
}
