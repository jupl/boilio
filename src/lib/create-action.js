import {createAction} from 'redux-actions'

/**
 * Same as redux-action's createAction function, but tacks on the type as
 * a property to the action reator.
 * @return {Function} Action creator
 */
export default function createActionMod(...args) {
  const actionCreator = createAction(...args)
  actionCreator.type = args[0]
  return actionCreator
}
