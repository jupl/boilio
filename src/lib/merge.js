/**
 * Merge properties for connect in react-redux. The key difference from the
 * default version is to place dispatch properties in its own property.
 * @param {?Object} props - Optional properties generated from state data
 * @param {?Object} actions - Optional properties generated with dispatch
 * @param {?Object} ownProps - Optional properties passed in from user
 * @return {Object} Properties to pass over to the component
 */
export default function merge(props, actions = {}, ownProps) {
  return {...ownProps, ...props, actions}
}
