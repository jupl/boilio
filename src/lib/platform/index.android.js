/** Platform OS as a proper noun */
export const NAME = 'Android'
