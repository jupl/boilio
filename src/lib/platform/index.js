/** Platform OS placeholder */
export const NAME = 'Testing'
