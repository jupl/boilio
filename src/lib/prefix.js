export default process.versions.node ? x => x : require('react-prefixer')
