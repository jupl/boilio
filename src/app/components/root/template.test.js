import React from 'react'
import {Text} from 'react-native'
import test from 'ava'
import {shallow} from 'enzyme'
import Root from './template'
import * as styles from './styles'
import * as platform from '../../../lib/platform'

/** @test {app.components.Root} */
// TODO Try to remove use of intermediate variable
test('<Root>', t => {
  const wrapper = shallow(<Root />)
  const contains = wrapper.contains(
    <Text style={styles.TEXT}>Welcome to {platform.NAME}</Text>
  )
  t.true(contains)
})
