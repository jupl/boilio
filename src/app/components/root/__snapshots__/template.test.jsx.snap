exports[`<App.Root> Template should render as expected 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#39cccc",
      "display": "flex",
      "flex": 1,
      "transition": "background 0.8s ease-out",
    }
  }>
  <div
    style={
      Object {
        "alignItems": "center",
        "background": "linear-gradient(rgba(255, 255, 255, 0.4), transparent)",
        "display": "flex",
        "flex": 1,
        "justifyContent": "center",
      }
    }>
    <button
      onClick={[Function]}
      style={
        Object {
          "background": "radial-gradient(white, gainsboro)",
          "borderColor": "gainsboro",
          "borderRadius": "50%",
          "outline": "none",
        }
      }>
      <
    </button>
    <span
      style={
        Object {
          "marginLeft": 4,
          "marginRight": 4,
        }
      }>
      Hello, World
    </span>
    <button
      onClick={[Function]}
      style={
        Object {
          "background": "radial-gradient(white, gainsboro)",
          "borderColor": "gainsboro",
          "borderRadius": "50%",
          "outline": "none",
        }
      }>
      >
    </button>
  </div>
</div>
`;
